### PR TITLE
mgr/mgr_module : Fix keyerror for op if rule steps first element is opcode

### PR DESCRIPTION
**Pull request by pdvian**
Fixes : https://tracker.ceph.com/issues/47272

Signed-off-by: Prashant Dhange <pdhange@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by pdvian** [comment]
Thanks @batrick. Let me correct it.

**Comment by pdvian** [comment]
The more simpler way would be :

`first_take = [s for s in rule['steps'] if 'op' in s and s['op'] == 'take'][0]`

**Comment by batrick** [comment]
`Fixes: ` whitespace needs fixed too please.

File: src/pybind/mgr/mgr_module.py [hunk 1]
```diff
@@ -234,7 +234,7 @@ def get_rule_root(self, rule_name):
             return None
 
         try:
-            first_take = [s for s in rule['steps'] if s['op'] == 'take'][0]
+            first_take = [s for s in rule['steps'] if 'op' in s and s['op'] == 'take'][0]
```
Comment (tchaikov) [inline src/pybind/mgr/mgr_module.py:237]:
```suggestion
        try:
            first_take = next(s for s in rule['steps'] if s.get('op') == 'take')
        except StopIteration:
```

**Review by batrick — APPROVED** [review]

**Comment by batrick** [comment]
jenkins test api

**Comment by tchaikov** [comment]
- https://pulpito.ceph.com/kchai-2020-09-06_13:32:55-rados-wip-kefu-testing-2020-09-06-2017-distro-basic-smithi/
- https://pulpito.ceph.com/kchai-2020-09-07_01:57:21-rados-wip-kefu-testing-2020-09-06-2017-distro-basic-smithi

failures tracked by 

- https://tracker.ceph.com/issues/47306
- https://tracker.ceph.com/issues/47179

and i tested locally using

```sh
../qa/run-standalone.sh osd-backfill-prio.sh
```

it passed.

**Review by tchaikov — APPROVED** [review]

**Pull request merged**